tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\SystemMailToUsersOfRoleTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+use Drupal\Core\Language\LanguageInterface;
+use Psr\Log\LogLevel;
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\user\Entity\User;
+use Drupal\user\Entity\Role;
+use Drupal\Tests\rules\Integration\RulesUserIntegrationTestTrait;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\RulesAction\SystemMailToUsersOfRole
+ * @group rules_actions
+ */
+class SystemMailToUsersOfRoleTest extends RulesEntityIntegrationTestBase {
+
+  use RulesUserIntegrationTestTrait;
+
+  /**
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * @var \Drupal\Core\Mail\MailManagerInterface
+   */
+  protected $mailManager;
+
+  /**
+   * @var \Drupal\user\UserStorage
+   */
+  protected $userStorage;
+
+  /**
+   * @var \Drupal\user\Entity\Role
+   */
+  protected $role;
+
+  /**
+   * @var \Drupal\user\Entity\User
+   */
+  protected $user;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Plugin\Action\SystemMailToUsersOfRole
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+
+    $this->logger = $this->getMock('Psr\Log\LoggerInterface');
+
+    $this->mailManager = $this->getMockBuilder('Drupal\Core\Mail\MailManagerInterface')
+      ->getMock();
+
+    // Mock the role entity.
+    $role_name = 'administrator';
+    $this->role = $this->getMockedUserRole($role_name);
+    $this->role->expects($this->any())
+      ->method('id')
+      ->willReturn($role_name);
```

Comment:
This should not be necessary, the ->getMockedUserRole() method of trait already does that?